packages/firestore/src/local/local_store.ts

```diff
@@ -345,7 +345,9 @@ export class LocalStore {
       'readwrite-primary',
       txn => {
         const affected = batchResult.batch.keys();
-        const documentBuffer = this.remoteDocuments.newChangeBuffer();
+        const documentBuffer = this.remoteDocuments.newChangeBuffer({
+          createSentinelDocumentsToTrackDeletes: true
```

Comment:
Part of me wonders if this should be the default behavior and GC should explicitly opt-out (e.g. flip the logic and rename to `suppressSentinelDeleteDocuments` or something) since reading this code, I would have trouble guessing what this is for and why it is enabled in this code path.  By contrast, the GC code could have a comment explaining why we don't want to track deletes in that case.

Alternatively, since this option probably corresponds 100% to whether this is GC code or not, we could consider specializing the name to match (e.g. `forGarbageCollection: true` or something)

WDYT?